operation_outcome.go

```diff
@@ -0,0 +1,31 @@
+package fhirclient
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/samply/golang-fhir-models/fhir-models/fhir"
+)
+
+type OperationOutcome struct {
+	fhir.OperationOutcome
+	ResourceType *string `bson:"resourceType" json:"resourceType"`
+}
+
+func (r OperationOutcome) IsOperationOutcome() bool {
+	if r.ResourceType == nil {
+		return false
+	}
+
+	return strings.EqualFold(*r.ResourceType, "OperationOutcome")
+}
+
+func (r OperationOutcome) Error() string {
+	var messages []string
+	for _, issue := range r.Issue {
+		if issue.Diagnostics != nil {
```

Comment:
diagnostics being an optional field (and details as well), don't we always want an entry for the particular operationoutcome? Even when diagnostics is not set? (and maybe support detail field as well). Or add a note/TODO that we currently don't support it ehre?